Haskell/Haskell.sublime-syntax

```diff
@@ -290,6 +295,21 @@ contexts:
       captures:
         1: punctuation.definition.comment.haskell
 
+  shebang:
+    - match: ^\#!
+      scope: punctuation.definition.comment.haskell
+      set: shebang-body
+    - match: ^|(?=\S)  # Note: Ensure to highlight shebang if Haskell is embedded.
+      pop: 1
+
+  shebang-body:
+    - meta_scope: comment.line.shebang.haskell
+    # Note: Keep sync with first_line_match!
```

Comment:
Probably not unique to this syntax, but I wonder whether we should make this match more general to allow for any executable instead of only highlighting the expected values and what is used for syntax detection? It could be a wrapper script, for example. A path that ends with `env` would then be exempted.

Reply:
I think a generic but still reliable approach which can be applied to all syntaxes would be welcome.

Reply:
> What do you think about a syntax or context in the Bash package that all other syntaxes could embed for their first line shebang match?

Does that still work if someone adds Bash to "ignored_packages"?

Reply:
It certainly won't. I also don't think it is required to `embed` such a tiny context. And we should probably try to keep dependencies between packages low. Inheritance comes with a certain level of complexity so a use case should be worth it.

Reply:
Sure it's not "required" but it does beg the question of how to deduplicate such a frequently occurring syntax element that would ideally be scoped consistently. We could of course also include a context, which would not produce errors, as far as I remember, though I'm not exactly sure what happens when you embed something that doesn't exist.

Reply:
An error about missing syntax is printed to console and the region up to the `escape` is unscoped.